receiver/windowsperfcountersreceiver/windowsperfcounters_scraper.go

```diff
@@ -45,40 +45,48 @@ type PerfCounterScraper interface {
 // scraper is the type that scrapes various host metrics.
 type scraper struct {
 	cfg      *Config
-	logger   *zap.Logger
-	counters []PerfCounterScraper
+	settings component.TelemetrySettings
+	counters []PerfCounterMetrics
 }
 
-func newScraper(cfg *Config, logger *zap.Logger) (*scraper, error) {
-	if err := cfg.Validate(); err != nil {
-		return nil, err
-	}
+type PerfCounterMetrics struct {
+	CounterScraper PerfCounterScraper
+	Attributes     map[string]string
+	Metric         string
+}
+
+type PerfCounterScrapedMetrics struct {
+	CounterScraper PerfCounterScraper
+	Attributes     map[string]string
+	Metric         string
+}
```

Comment:
It's not clear to me why these both exist.

Reply:
Removed extraneous one, had a different implementation that had the perfCounterScraper built into another struct and passed further down but flattened the implementation.